app/src/main/java/com/github/sdp/mediato/data/Database.java

```diff
@@ -123,6 +131,37 @@ public static CompletableFuture<User> getUser(String username) {
     return future;
   }
 
+    /**
+     * Retrieve a CompletableFuture containing the user
+     *
+     * @param email of the user to be retrieved
+     * @return CompletableFuture<User> containing the user object
+     */
+    public static CompletableFuture<User> getUserByEmail(String email) {
+        CompletableFuture<User> future = new CompletableFuture<>();
+        database.getReference(USER_PATH)
+                .orderByChild("email")
+                .equalTo(email)
+                .addValueEventListener(new ValueEventListener() {
+            @Override

```

Comment:
Is it possible to put this in a lambda expression ? The database class is probably going to be crowded. We should keep it as clean as possible.

Reply:
unfortunately, we have to pass a ValueEventListener to the addValueEventListener, so we cannot use a lambda here (I tried to shorten the number of lines needed for that section to keep it clean)